### PR TITLE
fix(downmerge): merge to x branch if same minor version and if current branch is release get branch version

### PR DESCRIPTION
**Pull request by abachyla**

### CODE REVIEW

File: .github/workflows/downmerge-releases.yml [hunk 1]
```diff
@@ -32,9 +32,17 @@ jobs:
...
           CURRENT_BRANCH="$(git branch --show-current)"
-          MAJOR_MINOR_PACKAGE_VERSION=$(echo $BASE_PACKAGE_VERSION | cut -d'.' -f1,2)
+
+          if [[ "$CURRENT_BRANCH" =~ "release" ]]; then
+            CURRENT_BRANCH_VERSION=$(echo $CURRENT_BRANCH | cut -d'/' -f2)
+            CURRENT_MAJOR_MINOR_PACKAGE_VERSION=$(echo $CURRENT_BRANCH_VERSION | cut -d'.' -f1,2)
+          else
+            BASE_PACKAGE_VERSION=$(node -p "require('./package.json').version")
+            CURRENT_MAJOR_MINOR_PACKAGE_VERSION=$(echo $BASE_PACKAGE_VERSION | cut -d'.' -f1,2)
+          fi
```
Comment (abachyla) [inline .github/workflows/downmerge-releases.yml:43]: this is the fix for blocking downmerge when x release branch is just created and not bumped. It happens on brands when we run `npm run prepare-release new` or `npm run prepare-release new major`

File: .github/workflows/downmerge-releases.yml [hunk 2]
```diff
@@ -32,9 +32,15 @@ jobs:
 
           MANUAL_MERGE=false
           # Determine the current branch and package version
-          BASE_PACKAGE_VERSION=$(node -p "require('./package.json').version")
           CURRENT_BRANCH="$(git branch --show-current)"
-          MAJOR_MINOR_PACKAGE_VERSION=$(echo $BASE_PACKAGE_VERSION | cut -d'.' -f1,2)
+
+          if [[ "$CURRENT_BRANCH" =~ "release" ]]; then
```
Comment (abachyla) [inline .github/workflows/downmerge-releases.yml:37]: this is the fix for blocking downmerge when x release branch is just created and not bumped. It happens on brands when we run npm run prepare-release new or npm run prepare-release new major

Comment (abachyla) [inline .github/workflows/downmerge-releases.yml:37]: this is the fix for blocking downmerge when x release branch is just created and not bumped. It happens on brands when we run `npm run prepare-release new` or `npm run prepare-release new major`

Comment (abachyla) [inline .github/workflows/downmerge-releases.yml:37]: this is the fix for blocking downmerge when x release branch is just created and not bumped. It happens on brands when we run `npm run prepare-release new` or `npm run prepare-release new major`

File: .github/workflows/downmerge-releases.yml [hunk 3]
```diff
@@ -46,9 +52,10 @@ jobs:
             BRANCH_VERSION=$(echo $branch | cut -d'/' -f2)
             MAJOR_MINOR_BRANCH_VERSION=$(echo $BRANCH_VERSION | cut -d'.' -f1,2)
             PATCH_BRANCH_VERSION=$(echo $BRANCH_VERSION | cut -d'.' -f3)
-            NEWER_VERSION="$(printf '%s\n' "$MAJOR_MINOR_BRANCH_VERSION" "$MAJOR_MINOR_PACKAGE_VERSION" | sort -V | tail -n 1)"
-            IS_CURRENT_RELEASE_BRANCH=$(if [[ $PATCH_BRANCH_VERSION == "x" && $MAJOR_MINOR_BRANCH_VERSION == $MAJOR_MINOR_PACKAGE_VERSION ]]; then echo "true"; else echo "false"; fi)
-            if [[ $branch == "develop" || $NEWER_VERSION == $MAJOR_MINOR_BRANCH_VERSION || $IS_CURRENT_RELEASE_BRANCH == true ]]; then
+            NEWER_VERSION="$(printf '%s\n' "$MAJOR_MINOR_BRANCH_VERSION" "$CURRENT_MAJOR_MINOR_VERSION" | sort -V | tail -n 1)"
+            IS_CURRENT_RELEASE_BRANCH=$(if [[ $PATCH_BRANCH_VERSION == "x" && $MAJOR_MINOR_BRANCH_VERSION == $CURRENT_MAJOR_MINOR_VERSION ]]; then echo "true"; else echo "false"; fi)
+
+            if [[ $branch == "develop" || ("$CURRENT_MAJOR_MINOR_VERSION" != "$MAJOR_MINOR_BRANCH_VERSION" && $NEWER_VERSION == $MAJOR_MINOR_BRANCH_VERSION) || $IS_CURRENT_RELEASE_BRANCH == true ]]; then
```
Comment (abachyla) [inline .github/workflows/downmerge-releases.yml:58]: if newer version is the current one, we should do downmerge only to x branch, $IS_CURRENT_RELEASE_BRANCH handles it

**Review by joseph118 — APPROVED** [review]

File: .github/workflows/downmerge-releases.yml [hunk 4]
```diff
@@ -32,9 +32,15 @@ jobs:
 
           MANUAL_MERGE=false
           # Determine the current branch and package version
-          BASE_PACKAGE_VERSION=$(node -p "require('./package.json').version")
           CURRENT_BRANCH="$(git branch --show-current)"
-          MAJOR_MINOR_PACKAGE_VERSION=$(echo $BASE_PACKAGE_VERSION | cut -d'.' -f1,2)
+
+          if [[ "$CURRENT_BRANCH" =~ "release" ]]; then
+            CURRENT_BRANCH_VERSION=$(echo $CURRENT_BRANCH | cut -d'/' -f2)
```
Comment (Lafaa) [inline .github/workflows/downmerge-releases.yml:38]: Call this `BASE_PACKAGE_VERSION` and bring the second statement outside the IF

Comment (abachyla) [reply .github/workflows/downmerge-releases.yml:38]: but if the branch is `master`, the value won't be valid

Comment (Lafaa) [reply .github/workflows/downmerge-releases.yml:38]: I explained myself better offline :P

**Review by Lafaa — APPROVED** [review]

**Pull request merged**